cmd/soroban-rpc/internal/events/events.go

```diff
@@ -208,7 +208,7 @@ func readEvents(networkPassphrase string, ledgerCloseMeta xdr.LedgerCloseMeta) (
 		}
 		for i := range tx.Envelope.Operations() {
 			opIndex := uint32(i)
-			var opEvents []xdr.ContractEvent
+			var opEvents []xdr.DiagnosticEvent
 			opEvents, err = tx.GetOperationEvents(opIndex)
```

Comment:
just to confirm for own understanding, `xdr.DiagnosticEvent` is always returned for events in meta now, even if `ENABLE_SOROBAN_DIAGNOSTIC_EVENTS=false` ?

Reply:
I think `ENABLE_SOROBAN_DIAGNOSTIC_EVENTS` refers to enabling the `Diagnostic` `ContractEventType`.

`DiagnosticEvent` seems to be used for both the global eventcontainer type and a contract event subtype. This is really confusing to be frank ...